Lib/wave.py

```diff
@@ -495,6 +499,13 @@ def open(f, mode=None):
             mode = f.mode
         else:
             mode = 'rb'
+    try:
+        f = os.fspath(f)
+    except TypeError:
+        if not hasattr(f, 'read') or not hasattr(f, 'close'):
+            raise TypeError('open() takes str, bytes, a PathLike object, '
+                            + f'or an open filehandle, not {type(f)}')
```

Comment:
`{type(f).__name__!r}` would give us a more readable name.

Reply:
Actually bytes are not supported with current implementation. See the comment above.